backend/audit/intakelib/checks/runners.py

```diff
@@ -146,43 +148,90 @@
+    If this is a data migration, then there are some checks we do not want to run.
+    If this is not a data migration, then we want to make sure no one put in a GSA_MIGRATION keyword.
+    """
     errors = []
+
+    errors += run_section_check(ir, section_name)
+    list_of_checks = filter_checks_for_data_migration(list_of_checks, is_data_migration)
+    check_for_gsa_migration_keyword_if_needed(ir, is_data_migration)
+
+    for fun in list_of_checks:
+        res = run_check(fun, ir, is_data_migration, auditee_uei)
+        errors += process_check_result(res)
+
+    log_errors(errors)
+    if errors:
+        raise ValidationError(errors)
+
+
+def run_section_check(ir, section_name):
+    """Run the section check if it is provided."""
     if section_name:
         res = is_right_workbook(section_name)(ir)
         if res:
-            errors.append(res)
-
-    # If this is a data migration, then there are some checks we do not want to run.
-    #
-    if is_data_migration:
-        if list_of_checks == federal_awards_checks:
-            list_of_checks = list(
-                filter(lambda f: f != check_cluster_names, list_of_checks)
-            )
-    else:
-        # This is not a data migration.
-        # We want to make sure no one put in a GSA_MIGRATION keyword.
+            return [res]
+    return []
+
+
+def filter_checks_for_data_migration(list_of_checks, is_data_migration):
+    """If this is a data migration, then there are some checks we do not want to run."""
+    if is_data_migration and list_of_checks == federal_awards_checks:
+        return list(filter(lambda f: f != check_cluster_names, list_of_checks))
+    return list_of_checks
+
+
+def check_for_gsa_migration_keyword_if_needed(ir, is_data_migration):
+    """If this is not a data migration, then we want to make sure no one put in a GSA_MIGRATION keyword."""
+    if not is_data_migration:
         check_for_gsa_migration_keyword(ir)
 
-    for fun in list_of_checks:
-        if fun == verify_auditee_uei_match:
-            res = fun(ir, auditee_uei)
-        else:
-            res = fun(ir)
-        if isinstance(res, list) and all(map(lambda v: isinstance(v, tuple), res)):
-            errors = errors + res
-        elif isinstance(res, tuple):
-            errors.append(res)
-        else:
-            pass
+
+def get_key_by_value(d, target_value):
+    keys = [key for key, value in d.items() if value == target_value]
+    return keys[0] if keys else None
+
+
+def run_check(fun, ir, is_data_migration, auditee_uei):
+    """Run the validation check if it is not skippable, or if it is skippable, only run if this is not a data migration."""
+    fun_name = get_key_by_value(skippable_checks, fun)
+    if fun == verify_auditee_uei_match:
+        return fun(ir, auditee_uei)
+    elif (
+        is_data_migration
+        and fun_name
+        and fun_name in InvalidRecord.fields["validations_to_skip"]
+    ):
+        return None
```

Comment:
This should be the first case, IMO

Reply:
The order would matter if they were related logic/conditions but in this case they are not.

Reply:
Discussed offline (readability)